pkg/webhook/resources/virtualmachinebackup/validator.go

```diff
@@ -56,9 +56,15 @@ func (v *virtualMachineBackupValidator) Create(request *types.Request, newObj ru
 		return werror.NewInvalidError("source VM name is empty", fieldSourceName)
 	}
 
-	_, err := v.vms.Get(newVMBackup.Namespace, newVMBackup.Spec.Source.Name)
-	if err != nil {
-		return werror.NewInvalidError(err.Error(), fieldSourceName)
+	var err error
+
+	// If VMBackup is from metadata in backup target, we don't check whether the VM is existent,
+	// because the related VM may not exist in a new cluster.
+	if newVMBackup.Status == nil {
```

Comment:
seems the status is set on the [createVMBackupIfNotExist](https://github.com/harvester/harvester/blob/26deb4cdb2d9dec90c6d3507b4b2fb08dc7d1022/pkg/controller/master/backup/backup_metadata.go#L163) method, wondering if it gets ignored automatically on the validator webhook?

Reply:
When we create a VMBackup from UI, it doesn't have `status`, so we ignore checking it before. However, a VMBackup from metadata has `status`, so we don't check the source VM here.

Reply:
thanks, seems the restoring on the new cluster is still having problems with:
```
handler harvester-vm-backup-controller: backups.longhorn.io \"backup-5a35fc06d8da48fd\" not found, requeuing"
```

```yaml
apiVersion: harvesterhci.io/v1beta1
kind: VirtualMachineBackup
metadata:
  creationTimestamp: "2022-11-08T04:01:20Z"
  finalizers:
  - wrangler.cattle.io/harvester-vm-backup-controller
  name: b2
  namespace: default
  resourceVersion: "22060545"
  uid: 9b38f17a-d01a-4eae-8a32-39b054a6ad40
spec:
  source:
    apiGroup: kubevirt.io
    kind: VirtualMachine
    name: ub-test
  type: backup
status:
  backupTarget:
    bucketName: backupbucket
    bucketRegion: us-east-1
    endpoint: https://minio-service.default:9000
  conditions:
  - lastTransitionTime: "2022-11-08T04:01:20Z"
    message: backups.longhorn.io "backup-5a35fc06d8da48fd" not found
    reason: Error
    status: "False"
    type: InProgress
  - lastTransitionTime: "2022-11-08T04:01:20Z"
    message: Not Ready
    status: "False"
    type: Ready
  error:
    message: backups.longhorn.io "backup-5a35fc06d8da48fd" not found
    time: "2022-11-08T04:11:50Z"
  readyToUse: false
  secretBackups:
  - data:
      userdata: I2Nsb3VkLWNvbmZpZwpwYWNrYWdlX3VwZGF0ZTogdHJ1ZQpwYWNrYWdlczoKICAtIHFlbXUtZ3Vlc3QtYWdlbnQKcnVuY21kOgogIC0gLSBzeXN0ZW1jdGwKICAgIC0gZW5hYmxlCiAgICAtIC0tbm93CiAgICAtIHFlbXUtZ3Vlc3QtYWdlbnQuc2VydmljZQo=
    name: ub-test-vltyo
  source:
    metadata:
      annotations:
        harvesterhci.io/vmRunStrategy: RerunOnFailure
        harvesterhci.io/volumeClaimTemplates: '[{"metadata":{"name":"ub-test-disk-0-glals","annotations":{"harvesterhci.io/imageId":"default/image-prsgs"}},"spec":{"accessModes":["ReadWriteMany"],"resources":{"requests":{"storage":"10Gi"}},"volumeMode":"Block","storageClassName":"longhorn-image-prsgs"}}]'
        kubevirt.io/latest-observed-api-version: v1
        kubevirt.io/storage-observed-api-version: v1alpha3
        network.harvesterhci.io/ips: '[]'
      creationTimestamp: null
      labels:
        harvesterhci.io/creator: harvester
        harvesterhci.io/os: ubuntu
      name: ub-test
      namespace: default
    spec:
      runStrategy: RerunOnFailure
      template:
        metadata:
          annotations:
            harvesterhci.io/sshNames: '[]'
          creationTimestamp: null
          labels:
            harvesterhci.io/vmName: ub-test
        spec:
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                - matchExpressions:
                  - key: network.harvesterhci.io/vlan
                    operator: In
                    values:
                    - "true"
          domain:
            cpu:
              cores: 2
              sockets: 1
              threads: 1
            devices:
              disks:
              - bootOrder: 1
                disk:
                  bus: virtio
                name: disk-0
              - disk:
                  bus: virtio
                name: cloudinitdisk
              inputs:
              - bus: usb
                name: tablet
                type: tablet
              interfaces:
              - bridge: {}
                macAddress: 52:69:4d:4d:f7:82
                model: virtio
                name: default
            features:
              acpi:
                enabled: true
            machine:
              type: q35
            memory:
              guest: 3996Mi
            resources:
              limits:
                cpu: "2"
                memory: 4Gi
              requests:
                cpu: 125m
                memory: 2730Mi
          evictionStrategy: LiveMigrate
          hostname: ub-test
          networks:
          - multus:
              networkName: default/vlan1238
            name: default
          volumes:
          - name: disk-0
            persistentVolumeClaim:
              claimName: ub-test-disk-0-glals
          - cloudInitNoCloud:
              networkDataSecretRef:
                name: ub-test-vltyo
              secretRef:
                name: ub-test-vltyo
            name: cloudinitdisk
  volumeBackups:
  - csiDriverName: driver.longhorn.io
    longhornBackupName: backup-5a35fc06d8da48fd
    name: b2-volume-ub-test-disk-0-glals
    persistentVolumeClaim:
      metadata:
        annotations:
          harvesterhci.io/imageId: default/image-prsgs
          harvesterhci.io/owned-by: '[{"schema":"kubevirt.io.virtualmachine","refs":["default/ub-test"]}]'
          pv.kubernetes.io/bind-completed: "yes"
          pv.kubernetes.io/bound-by-controller: "yes"
          volume.beta.kubernetes.io/storage-provisioner: driver.longhorn.io
          volume.kubernetes.io/storage-provisioner: driver.longhorn.io
        creationTimestamp: null
        name: ub-test-disk-0-glals
        namespace: default
      spec:
        accessModes:
        - ReadWriteMany
        resources:
          requests:
            storage: 10Gi
        storageClassName: longhorn-image-prsgs
        volumeMode: Block
        volumeName: pvc-af1a42d8-c7dd-494f-a09d-9efec1d6c103
    volumeName: disk-0
```

Reply:
May you check whether you have related LH Backup in your backup-target? LH Backup is restored by LH controller.

Reply:
I fixed the error. We don't reset LH backup-target when the setting value is empty. In this case, if we delete vmbackup, the related LH backup will be deleted too.

https://github.com/harvester/harvester/pull/3107/commits/0719b8a73ddb9e925d5f78ca86ea0c947363ea54